actionpack/lib/action_controller/metal/rendering.rb

```diff
@@ -78,7 +78,9 @@ def _set_rendered_content_type(format)
       end
 
       def _set_vary_header
-        self.headers["Vary"] = "Accept" if request.should_apply_vary_header?
+        if self.headers["Vary"].blank? && request.should_apply_vary_header?
```

Comment:
`request.should_apply_vary_header?`  itself performs multiple checks for adding the header or not. I think it'd be more consistent if we put this check into it as well?